### PR TITLE
[TE] ThirdEye User Configuration Validation - detection and subscription yaml

### PR DESCRIPTION
**Pull request by akshayrai**
ThirdEye uses YAML files to manage the configuration of alerts and subscription groups. This approach enables employees across engineering and non-engineering, to configure simple to complex rules in an efficient way. However, mistakes in such files are easily made. These mistakes could lead to detection/subscription errors and fail to detect & notify a critical anomaly.

This PR tackles and addresses this issue by introducing schema based validations on user configs. I have used the json-schema-validator to achieve this. You will find the validation schema defined in `detection-config-schema.json` and `subscription-config-schema.json`

Changes:
* Refactored the validation interfaces with - static and semantic validation
* Introduced the json schema based validation of detection and subscription config

Tests:
* Added unit testing of configs
* Deployed and tested locally if the validation messages are displayed to the user

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/resources/validators/detection/detection-config-schema.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    },
+
+    "eachComponent": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": [ "type", "name" ],
+        "properties": {
+          "type": {
+            "type": "string"
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/resources/validators/detection/detection-config-schema.json:156]: shall the rule type be an enum?

File: thirdeye/thirdeye-pinot/src/main/resources/validators/subscription/subscription-config-schema.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  "properties": {
+    "subscriptionGroupName": {
+      "type": "string"
+    },
+    "application": {
+      "type": "string"
+    },
+    "type": {
+      "type": "string"
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/resources/validators/subscription/subscription-config-schema.json:17]: shall the subscription group type be an enum type?

File: thirdeye/thirdeye-pinot/src/main/resources/validators/detection/detection-config-schema.json (hunk 1 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/resources/validators/detection/detection-config-schema.json:156]: We can define an enum, but I avoided it for a couple of reasons.
1. We do not want to hard code and maintain the list of possible rule types here.
2. We have internal algorithm types which shouldn't be hardcoded here.
3. We run a validation in the code which will anyways capture if the type is invalid.

File: thirdeye/thirdeye-pinot/src/main/resources/validators/subscription/subscription-config-schema.json (hunk 2 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/resources/validators/subscription/subscription-config-schema.json:17]: replied above.

**Pull request merged**